pubsub/drivertest/drivertest.go

```diff
@@ -389,3 +397,67 @@ func testAs(t *testing.T, newHarness HarnessMaker, st AsTest) {
 		t.Error(err)
 	}
 }
+
+// Publishes a large number of messages to topic concurrently, and then times
+// how long it takes to receive them all.
+func benchmarkReceive(b *testing.B, topic *pubsub.Topic, sub *pubsub.Subscription) {
+	attrs := map[string]string{"label": "value"}
+	body := []byte("hello, world")
+	const (
+		nMessages          = 10000
+		concurrencySend    = 100
+		concurrencyReceive = 1
+	)
+	if nMessages%concurrencySend != 0 || nMessages%concurrencyReceive != 0 {
+		b.Fatal("nMessages must be divisible by # of sending/receiving goroutines")
+	}
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		b.StopTimer()
```

Comment:
Why is this here instead of at the bottom of the loop?

Reply:
FYI, this code is copied from the previous benchmark, which was GCPPubsub-specific.

The timer is stopped during the publish phase, because the benchmark is specifically timing `Receive` and not `Send`. So we `StopTimer` before publishing, and `StartTimer` again before receive.